cdap-watchdog/src/main/java/co/cask/cdap/logging/framework/LogFileOutputStream.java

```diff
@@ -85,14 +87,22 @@ long getCreateTime() {
     return createTime;
   }
 
+  /**
+   * get the number of bytes written to output stream
+   * @return file size
+   */
+  long getSize() {
+    return fileSize;
+  }
+
   @Override
   public void flush() throws IOException {
     dataFileWriter.flush();
   }
 
   @Override
   public void sync() throws IOException {
-    flush();
+    fileSize = dataFileWriter.sync();
```

Comment:
@chtyim should we also call flush ? i noticed that flush in `DataFileWriter`only calls the output stream's flush in addition to what it does in `sync`, which we do (flushing outputstream) in this method anyway. so i think calling sync is enough. let me know if its not.